### PR TITLE
[RLlib; fault-tolerance] Fix spot node preemption problem (RLlib does not catch correct `ObjectLostError`).

### PR DESCRIPTION
**Pull request by sven1977**
Fix spot node preemption problem (RLlib does not catch correct `ObjectLostError`).

If EnvRunners are located on a SPOT instance and the node pre-empts, sometimes an `ObjectLostError` is thrown from the `ray.get` call inside RLlib's `FaultTolerantActorManager`. This error wouldn't be properly handled and thus the main process (the owner of the `FaultTolerantActorManager`) would crash.

This would happen even if all proper fault-tolerance settings have been applied:
```
config.fault_tolerance(ignore_env_runner_failures=True OR recreate_failed_env_runners=True)
```

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/BUILD [hunk 1]
```diff
@@ -959,12 +952,20 @@ py_test(
...
     size = "large",
-    srcs = ["algorithms/tests/test_worker_failures.py"]
+    srcs = ["algorithms/tests/test_env_runner_failures.py"]
+)
+
+py_test(
+    name = "test_node_failures",
+    tags = ["team:rllib", "tests_dir", "exclusive"],
+    size = "medium",
+    srcs = ["algorothms/tests/test_node_failure.py"],
```
Comment (simonsays1980) [inline rllib/BUILD:965]: "algorothms" -> "algorithms" ;)

**Review by simonsays1980 — APPROVED** [review]
LGTM. This is a huge thing. Preemptive instances! Thanks so much for this fix @sven1977. One thing: Could we add a hint in the docstrings in `AlgorithmConfig` to `recreate_failed_env_runners` that this has to be `True` for spot instances, please?

**Comment by sven1977** [comment]
Thanks for your review @simonsays1980 . Will add the requested doc change. I'm thinking of even switching to `recreate_failed_env_runners=True` by default. I think it's the setting that now makes most sense. All algos behave very well upon EnvRunner failure and restart now, so this should not pose any problems.

Will create a follow-up PR with doc changes and this change.

File: rllib/BUILD (hunk 1 above)
Comment (sven1977) [reply rllib/BUILD]: fixed

**Comment by sven1977** [comment]
> LGTM. This is a huge thing. Preemptive instances! Thanks so much for this fix @sven1977. One thing: Could we add a hint in the docstrings in `AlgorithmConfig` to `recreate_failed_env_runners` that this has to be `True` for spot instances, please?

Done.

**Pull request merged**